aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -178,15 +178,14 @@ void copy_cast_mps(at::Tensor& dst, const at::Tensor& src,
   id<MTLBuffer> destBuffer = getMTLBufferStorage(dst_);
   uint64_t src_total_size = 0;
 
-  // This is weird, but sometimes this function can be called
-  //  with contiguous destination and non-contiguous source
-  if (src_.is_view() || dst_.is_contiguous() != src_.is_contiguous()) {
+  // Get contiguous view if src and dst tensors are differently strided
+  if (src_.strides() != dst_.strides()) {
```

Comment:
I am curious if this is a better condition to check:

```
if (src_.is_view() || src_.strides() != dst_.strides())

Reply:
Scratch above it looks fine.

Reply:
Note that `.is_view()` is a autograd-only concept. It only reflect if the Tensor is a backward AD differentiable view of another Tensor.
It has nothing to do with data sharing with other Tensors or striding of the Tensor.